vm/src/builtins/object.rs

```diff
@@ -184,8 +185,23 @@ impl PyBaseObject {
 
     /// Return repr(self).
     #[pymethod(magic)]
-    fn repr(zelf: PyObjectRef) -> String {
-        format!("<{} object at {:#x}>", zelf.class().name, zelf.get_id())
+    fn repr(zelf: PyObjectRef, vm: &VirtualMachine) -> String {
+        let class = zelf.class();
+        match class.module(vm).downcast_ref::<PyStr>().map(|m| m.as_str()) {
+            Some(module) if module != "builtins" => {
+                format!(
+                    "<{}.{} object at {:#x}>",
+                    module,
+                    class
+                        .qualname(vm)
+                        .downcast_ref::<PyStr>()
+                        .map(|n| n.as_str())
+                        .unwrap_or_else(|| class.name.as_str()),
+                    zelf.get_id()
+                )
+            }
+            _ => format!("<{} object at {:#x}>", class.tp_name(), zelf.get_id()),
+        }
```

Comment:
```suggestion
    format!("<{} object at {:#x}>", class.tp_name(), zelf.get_id())
```
So this code must be same as now